sdk/nodejs/helm/v2/helm.ts

```diff
@@ -233,77 +233,27 @@ export class Chart extends yaml.CollectionComponentResource {
         // overrides its predecessory, rather than throwing an exception, and `schema:
         // jsyaml.CORE_SCHEMA` to avoid using additional YAML parsing rules not supported by the
         // YAML parser used by Kubernetes.
-        const objs = yamlStream.split(/^---/m)
-            .map(yaml => jsyaml.safeLoad(yaml, {json: true, schema: jsyaml.CORE_SCHEMA}))
-            .filter(a => a != null && "kind" in a)
-            .sort(helmSort);
-        return yaml.parse(
-            {
-                resourcePrefix: resourcePrefix,
-                yaml: objs.map(o => jsyaml.safeDump(o)),
-                transformations: transformations || [],
-            },
-            { parent: this, dependsOn: dependsOn }
-        );
-    }
-}
-
-// helmSort is a JavaScript implementation of the Helm Kind sorter[1]. It provides a
-// best-effort topology of Kubernetes kinds, which in most cases should ensure that resources
-// that must be created first, are.
-//
-// [1]: https://github.com/helm/helm/blob/094b97ab5d7e2f6eda6d0ab0f2ede9cf578c003c/pkg/tiller/kind_sorter.go
-/** @ignore */ export function helmSort(a: { kind: string }, b: { kind: string }): number {
-    const installOrder = [
-        "Namespace",
-        "ResourceQuota",
-        "LimitRange",
-        "PodSecurityPolicy",
-        "Secret",
-        "ConfigMap",
-        "StorageClass",
-        "PersistentVolume",
-        "PersistentVolumeClaim",
-        "ServiceAccount",
-        "CustomResourceDefinition",
-        "ClusterRole",
-        "ClusterRoleBinding",
-        "Role",
-        "RoleBinding",
-        "Service",
-        "DaemonSet",
-        "Pod",
-        "ReplicationController",
-        "ReplicaSet",
-        "Deployment",
-        "StatefulSet",
-        "Job",
-        "CronJob",
-        "Ingress",
-        "APIService"
-    ];
-
-    const ordering: { [key: string]: number } = {};
-    installOrder.forEach((_, i) => {
-        ordering[installOrder[i]] = i;
-    });
-
-    const aKind = a["kind"];
-    const bKind = b["kind"];
-
-    if (!(aKind in ordering) && !(bKind in ordering)) {
-        return aKind.localeCompare(bKind);
-    }
-
-    if (!(aKind in ordering)) {
-        return 1;
-    }
-
-    if (!(bKind in ordering)) {
-        return -1;
+        const promise = pulumi.output(pulumi.runtime.invoke(
+            "kubernetes:yaml:parse",
+            {input: [{name: "helm", type: "yaml", value: yamlStream}]},
```

Comment:
It's definitely a shame to have to realize the whole `yamlStream` on the JS side and then pass it to the Go side.  It would be nice if we could move the `helm template` call itself into the Go side so that this whole buffer only need to be realized once.

Reply:
Yes, that's my plan. I was hoping to do that in a follow up PR that builds helm support into the provider